scipy/optimize/zeros.c

```diff
@@ -6,6 +6,48 @@
 
 #include "Python.h"
 #include <setjmp.h>
+#include <numpy/npy_math.h>
+
+//#include <stdio.h>
+#if PY_VERSION_HEX >= 0x03000000
+    #define PyString_FromString PyBytes_FromString
+    #define PyString_Concat PyBytes_Concat
+    #define PyString_AsString PyBytes_AsString
+    #define PyInt_FromLong PyLong_FromLong
+    #define PyInt_AsLong PyLong_AsLong
+#endif
+
+#ifndef FALSE
+#define FALSE 0
+#endif /* FALSE */
+#ifndef TRUE
+#define TRUE 1
+#endif /* TRUE */
+
+#include "ccallback.h"
+#include "Zeros/zeros.h"
+
+/*
+ * Caller entry point functions
+ */
+
+typedef enum {
+    CB_D = 1,         // f(x)
+    CB_D_D = 2,       // f(x, double)
+    CB_D_I_ND = 10,   // f(x, int n, double *)   I.e. pointer to n doubles
+    CB_D_VOIDSTAR = 100,     // f(x, void *) or f(x, const void *)
```

Comment:
Is there a clear need to add `CB_D_D`, `CB_D_I_ND`, `CB_D_VOIDSTAR`?
If not, I would not add them, for simplicity.

The quad() interface does have this sort of handling for `args=` piggypacking on top of the n-dim integration code, but that's mainly for backward compatibility, and I think the complexity it adds is not really well-justified.

Reply:
I had started from the list `quadpack_call_signatures` but after implementing these I agree that not all are necessary.